eth_bridge/ic/src/eth_proxy/src/common/types.rs

```diff
@@ -77,7 +80,7 @@ pub struct ProxyState {
     /// store incoming messages against status locks
     pub incoming_messages: RefCell<HashMap<MessageHash, MessageStatus>>,
     /// user balances
-    pub balances: RefCell<HashMap<Principal, HashMap<TokendId, Nat>>>,
+    pub balances: RefCell<HashMap<Principal, Vec<(Principal, Nat)>>>,
```

Comment:
```suggestion
    pub balances: RefCell<HashMap<EthereumAddr, Nat>>,
```
Could we narrow it down to this?

Reply:
If we need `caller` then ignore

Reply:
we need the caller